### ISSUE TITLE
consider dropping JSONObject from the smarthome API surface

### ISSUE THREAD

**Issue by proppy** [actions-on-google/actions-on-google-java#43]
Currently the libraries exposes JSONObject as part of the public smarthome API object:
https://github.com/actions-on-google/actions-on-google-java/blob/7f3b1e2fd51e3b044380ea3c72722058ad99240f/src/main/kotlin/com/google/actions/api/smarthome/SmartHomeResponse.kt

This force reverse dependencies to both depens on `org.json` (as a direct dependency) and `gson` (transitively).

It would be nice if the library instead only surfaced standard `Map<String, Object>` types in the public API.

/cc @taycaldwell @Fleker

**Comment by Fleker** [comment]
Yeah we should keep this in mind for the future, although it would end up being a breaking change.
